### PR TITLE
TCES-30 urd import clients and employers

### PR DESCRIPTION
**Pull request by tsids**
Created the import csv file page.

Completed TCES-30.

image

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 // import CreatePage from "./pages/create-user/create-user";
 // import EditPage from "./pages/edit-user/edit-user";
+import ImportPage from "./pages/import/import";
 
 function App() {
   return (
     <div className="App">
-      <DashboardPage />
+      {/* <DashboardPage /> */}
+      <ImportPage />
```
Comment (DanielDervishi) [inline client/src/App.jsx]: Revert changes

File: client/src/components/import-component/import-component.jsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  const [menu, setMenu] = useState("");
+  const [files, setFiles] = useState([]);
+  const [progress, setProgress] = useState(0);
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setProgress((oldProgress) => {
+        // if (oldProgress === 100) {
+        //   return 0;
+        // }
```
Comment (DanielDervishi) [inline client/src/components/import-component/import-component.jsx]: remove comment

File: client/src/components/import-component/import-component.jsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                  </Grid>
+                  <Grid item xs>
+                    <Stack direction="column" spacing={2}>
+                      <Box>
+                        <Typography variant="subtitle1">{file.name}</Typography>
+                        <Typography
+                          variant="body2"
+                          style={{ color: "rgba(0, 0, 0, 0.60)" }}
+                        >
+                          {file.size}kb •{" "}
```
Comment (DanielDervishi) [inline client/src/components/import-component/import-component.jsx]: I recommend using https://www.npmjs.com/package/filesize just to assure that the filesize isn't always in kb

File: client/src/pages/import/import.jsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import ImportComponent from "../../components/import-component/import-component";
```
Comment (jordanjanakievski) [inline client/src/pages/import/import.jsx]: Same thing for this file, naming it index.jsx

File: client/src/components/import-component/import-component.jsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+function importComponent() {
+  const [menu, setMenu] = useState("");
+  const [files, setFiles] = useState([]);
+  const [progress, setProgress] = useState(0);
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setProgress((oldProgress) => {
+        // if (oldProgress === 100) {
```
Comment (jordanjanakievski) [inline client/src/components/import-component/import-component.jsx]: We should remove commented out code chunks

File: client/src/components/import-component/import-component.jsx [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+      clearInterval(timer);
+    };
+  }, []);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    // Grab the file
+    const { files: currFiles } = e.target.files;
+    // Create a new FormData object
+    const formData = new FormData();
```
Comment (jordanjanakievski) [inline client/src/components/import-component/import-component.jsx]: Does this handle reading the file and converting it to an acceptable JSON that can be passed to the backend? The bulk addition of object will require the frontend to handle the conversion of the CSV to an acceptable list of JSON objects.

File: client/src/components/import-component/import-component.jsx [hunk 7]
```diff
@@ -0,0 +1,203 @@
+import { useState, useEffect, useRef } from "react";
```
Comment (jordanjanakievski) [inline client/src/components/import-component/import-component.jsx]: Let's rename this file index.jsx so it is uniform with the rest of the codebase

File: client/src/App.jsx [hunk 8]
```diff
@@ -1,13 +1,15 @@
 import "./App.css";
-import DashboardPage from "./pages/dashboard";
+// import DashboardPage from "./pages/dashboard";
```
Comment (DanielDervishi) [inline client/src/App.jsx]: Here's what happens when I try to upload many fies.
[screen-capture.webm](https://github.com/uoftblueprint/tces/assets/58835213/553bd154-e137-4af1-9f90-a358aa52b8df)

**Review by jordanjanakievski — COMMENTED** [review]
Just a couple comments, otherwise it's looking really good.
Request a review when ready.

Thank you!

File: client/src/components/import-component/import-component.jsx (hunk 6 above)
Comment (jordanjanakievski) [reply client/src/components/import-component/import-component.jsx]: We have a meeting with Wilfred tomorrow and we will be able to get some specific information on that data that can be expected. We will provide some of the data examples

File: client/src/components/import-component/index.jsx [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+    };
+  }, []);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    // Create a new FormData object
+    const formData = new FormData();
+    // Add the file(s) to the FormData object
+    formData.append("files", files);
+
```
Comment (DanielDervishi) [inline client/src/components/import-component/index.jsx]: Create a method for parsing the csvs that returns a list of objects -> one object for each row of the csv

**Review by DanielDervishi — COMMENTED** [review]
Great work once again taha!
I've just left a few comments.

File: client/src/App.jsx (hunk 8 above)
Comment (tsids) [reply client/src/App.jsx]: I can't see the video, but it works fine on my end.

Comment (jordanjanakievski) [reply client/src/App.jsx]: I think what Daniel is demonstrating that if they try to upload too many files at once, it breaks the frontend. Maybe we can limit how many files can be uploaded at once. @DanielDervishi let me know if this is what you might have been trying to show

Comment (DanielDervishi) [reply client/src/App.jsx]: yep this is what I was trying to show. You can replicate this by uploading many csvs

**Comment by tsids** [comment]
I've set a limit of max 4 files to be uploaded/selected at once (this is because >= 5 files ruins the layout of the page).
Also, I set a limit of 1 MB on the files.
Should I increase this limit, or leave it as is?

**Comment by DanielDervishi** [comment]
> I've set a limit of max 4 files to be uploaded/selected at once (this is because >= 5 files ruins the layout of the page). Also, I set a limit of 1 MB on the files. Should I increase this limit, or leave it as is?

Its probably best to increase the limit. We don't really know how many records TCES has at the moment. Also is there a reason you are setting a limit on the file sizes at the moment?

File: client/src/components/import-component/index.jsx [hunk 10]
```diff
@@ -0,0 +1,281 @@
+import { useState, useEffect, useRef } from "react";
```
Comment (DanielDervishi) [inline client/src/components/import-component/index.jsx:1]: image
The bottom of the page seems to be slightly messed up when I upload 4 files. Also I can't scroll all the way up.

**Review by DanielDervishi — COMMENTED** [review]
Great work Taha! It's almost there, just a few styling changes and it should be good to merge in!

**Review by jordanjanakievski — APPROVED** [review]
Good work! Everything is looking good and thanks for problem-solving with us!

**Review by DanielDervishi — APPROVED** [review]
Great Work Taha!

**Comment by tsids** [comment]
Should I merge the changes?

**Pull request merged**